src/plugins/controls/public/control_group/component/control_frame_component.tsx

```diff
@@ -149,18 +102,13 @@ export const ControlFrame = ({
         </>
       }
     >
-      {embeddable && !fatalError && (
+      {embeddable && (
         <div
           className={embeddableParentClassNames}
           id={`controlFrame--${embeddableId}`}
           ref={embeddableRoot}
         >
-          {fatalError && <ControlFrameError error={fatalError} />}
-        </div>
-      )}
-      {fatalError && (
-        <div className={embeddableParentClassNames} id={`controlFrame--${embeddableId}`}>
-          {<ControlFrameError error={fatalError} />}
+          {isErrorEmbeddable(embeddable) && <ControlError error={embeddable.error} />}
```

Comment:
The previous logic was overcomplicated + it didn't render properly because the following `ControlFrameError` component could never be rendered:

```typescript
     {embeddable && !fatalError && (
        <div
          className={embeddableParentClassNames}
          id={`controlFrame--${embeddableId}`}
          ref={embeddableRoot}
        >
          {fatalError && <ControlFrameError error={fatalError} />} // conflicts with !fatalError condition above
        </div>
      )}
```

![Screenshot 2023-05-23 at 12 08 23 PM](https://github.com/elastic/kibana/assets/8698078/6b1640c5-5066-4bff-81c1-8c7818a1bf43)